### ISSUE TITLE
AM2 Production Gone

### ISSUE THREAD

**Issue by BurningChromosome** [MagicBees/MagicBees#81]
I switched from 2.3.1 to 2.3.2 because of [#67](https://github.com/MagicBees/MagicBees/issues/67), but then all AM2 production was gone.

**Comment by MysteriousAges** [comment]
Not sure what you mean by "production was gone". Can you elaborate, please? What exactly is gone, and which bees do you have working?

**Comment by BurningChromosome** [comment]
While running 2.3.1 one can produce many of the Ars Magica 2 essences through magic bees (see image AM2-MB-A-01 - AM2-MB-A-11). In 2.3.2 the same bees do not produce any essences (see image AM2-MB-B-01 - AM2-MB-B-11). I also tested the Aqueous bee in an alveary with maximum "Frame Housing" blocks and impregnated frames. It produced 44 combs in one life-cycle, but no water essence (see image AM2-MB-C-01 - AM2-MB-C-04).

I have also included a list of all the mods I am running on the server

![am2-mb-a-01](https://cloud.githubusercontent.com/assets/13392074/8885192/d2fba9ca-325d-11e5-8384-e105e0cdc72c.png)
am2-mb-a-01

![am2-mb-a-02](https://cloud.githubusercontent.com/assets/13392074/8885193/d30047d2-325d-11e5-9c97-9992da0a2fdf.png)
am2-mb-a-02

![am2-mb-a-03](https://cloud.githubusercontent.com/assets/13392074/8885196/d306d9da-325d-11e5-8a4b-a49849eb7e52.png)
am2-mb-a-03

![am2-mb-a-04](https://cloud.githubusercontent.com/assets/13392074/8885194/d304e7ba-325d-11e5-9256-4b11e7eca6f4.png)
am2-mb-a-04

![am2-mb-a-05](https://cloud.githubusercontent.com/assets/13392074/8885198/d30c2bf6-325d-11e5-9122-8ef91a6c140d.png)
am2-mb-a-05

![am2-mb-a-06](https://cloud.githubusercontent.com/assets/13392074/8885197/d3087e20-325d-11e5-89a6-2dc1be653252.png)
am2-mb-a-06

![am2-mb-a-07](https://cloud.githubusercontent.com/assets/13392074/8885199/d3176264-325d-11e5-9e5b-4827bbbb7bbc.png)
am2-mb-a-07

![am2-mb-a-08](https://cloud.githubusercontent.com/assets/13392074/8885200/d31bc6d8-325d-11e5-8296-77dd49c4bafa.png)
am2-mb-a-08

![am2-mb-a-09](https://cloud.githubusercontent.com/assets/13392074/8885201/d321d9ce-325d-11e5-9dac-dc8efcf3f12a.png)
am2-mb-a-09

![am2-mb-a-10](https://cloud.githubusercontent.com/assets/13392074/8885202/d322fbec-325d-11e5-80d9-994fc23d786e.png)
am2-mb-a-10

![am2-mb-a-11](https://cloud.githubusercontent.com/assets/13392074/8885203/d325ca34-325d-11e5-8847-ca55a7ec1c7c.png)
am2-mb-a-11

![am2-mb-b-01](https://cloud.githubusercontent.com/assets/13392074/8885204/d32a2bc4-325d-11e5-916f-df9e971c265b.png)
am2-mb-b-01

![am2-mb-b-02](https://cloud.githubusercontent.com/assets/13392074/8885205/d332d47c-325d-11e5-9a9e-97c8de5997bf.png)
am2-mb-b-02

![am2-mb-b-03](https://cloud.githubusercontent.com/assets/13392074/8885206/d3376474-325d-11e5-8d65-0597f83f4f4d.png)
am2-mb-b-03

![am2-mb-b-04](https://cloud.githubusercontent.com/assets/13392074/8885207/d33d569a-325d-11e5-8556-32bb88f9c61e.png)
am2-mb-b-04

![am2-mb-b-05](https://cloud.githubusercontent.com/assets/13392074/8885208/d34155a6-325d-11e5-95b4-0dcb8c452fa1.png)
am2-mb-b-05

![am2-mb-b-06](https://cloud.githubusercontent.com/assets/13392074/8885209/d3458cd4-325d-11e5-8cbb-06fde7a83d0c.png)
am2-mb-b-06

![am2-mb-b-07](https://cloud.githubusercontent.com/assets/13392074/8885210/d347a370-325d-11e5-8f17-aa059b439dd0.png)
am2-mb-b-07

![am2-mb-b-08](https://cloud.githubusercontent.com/assets/13392074/8885211/d34ec25e-325d-11e5-9939-ed360b105828.png)
am2-mb-b-08

![am2-mb-b-09](https://cloud.githubusercontent.com/assets/13392074/8885213/d35a0c40-325d-11e5-9ac9-a3ecdbf67076.png)
am2-mb-b-09

![am2-mb-b-10](https://cloud.githubusercontent.com/assets/13392074/8885214/d35cfaa4-325d-11e5-8c4a-c2481f4f4356.png)
am2-mb-b-10

![am2-mb-b-11](https://cloud.githubusercontent.com/assets/13392074/8885215/d35f16b8-325d-11e5-9fb1-0f8969f9925c.png)
am2-mb-b-11

![am2-mb-c-01](https://cloud.githubusercontent.com/assets/13392074/8885216/d361d196-325d-11e5-8e64-dfd95ffc3b7e.png)
am2-mb-c-01

![am2-mb-c-02](https://cloud.githubusercontent.com/assets/13392074/8885217/d365bb76-325d-11e5-8452-248f662159ae.png)
am2-mb-c-02

![am2-mb-c-03](https://cloud.githubusercontent.com/assets/13392074/8885218/d36a6d4c-325d-11e5-9ec8-c7df6bcf4c10.png)
am2-mb-c-03

![am2-mb-c-04](https://cloud.githubusercontent.com/assets/13392074/8885219/d374ddae-325d-11e5-9dba-57ee877480df.png)
am2-mb-c-04

	ML: MCP v9.05 FML v7.10.99.99 Minecraft Forge 10.13.4.1448 69 mods loaded, 69 mods active
	UCHIJAAAA	mcp{9.05} [Minecraft Coder Pack] (minecraft.jar) 
	UCHIJAAAA	FML{7.10.99.99} [Forge Mod Loader] (FTBServer-1.7.10-1448.jar) 
	UCHIJAAAA	Forge{10.13.4.1448} [Minecraft Forge] (FTBServer-1.7.10-1448.jar) 
	UCHIJAAAA	AM2-Preloader{0.0.2} [AMCore] (minecraft.jar) 
	UCHIJAAAA	CodeChickenCore{1.0.6.43} [CodeChicken Core] (minecraft.jar) 
	UCHIJAAAA	ThaumicTinkerer-preloader{0.1} [Thaumic Tinkerer Core] (minecraft.jar) 
	UCHIJAAAA	<CoFH ASM>{000} [CoFH ASM] (minecraft.jar) 
	UCHIJAAAA	AnimationAPI{1.2.4} [AnimationAPI] (AnimationAPI-1.7.10-1.2.4.jar) 
	UCHIJAAAA	arsmagica2{1.4.0.008} [Ars Magica 2] (1.7.10_AM2-1.4.0.008.jar) 
	UCHIJAAAA	ArchimedesShips{1.7.10 v1.7.1} [Archimedes' Ships] (ArchimedesShips-1.7.1.jar) 
	UCHIJAAAA	BiblioCraft{1.10.4} [BiblioCraft] (BiblioCraft[v1.10.4][MC1.7.10].jar) 
	UCHIJAAAA	Mantle{1.7.10-0.3.2.jenkins184} [Mantle] (Mantle-1.7.10-0.3.2.jar) 
	UCHIJAAAA	Natura{2.2.0} [Natura] (natura-1.7.10-2.2.0.1.jar) 
	UCHIJAAAA	BiomesOPlenty{2.1.0} [Biomes O' Plenty] (BiomesOPlenty-1.7.10-2.1.0.1067-universal.jar) 
	UCHIJAAAA	BiblioWoodsBoP{1.9} [BiblioWoods Biomes O'Plenty Edition] (BiblioWoods[BiomesOPlenty][v1.9].jar) 
	UCHIJAAAA	CoFHCore{1.7.10R3.0.2} [CoFH Core] (CoFHCore-[1.7.10]3.0.2-262.jar) 
	UCHIJAAAA	Forestry{3.5.7.16} [Forestry for Minecraft] (forestry_1.7.10-3.5.7.16 (1).jar) 
	UCHIJAAAA	BiblioWoodsForestry{1.7} [BiblioWoods Forestry Edition] (BiblioWoods[Forestry][v1.7].jar) 
	UCHIJAAAA	BiblioWoodsNatura{1.5} [BiblioWoods Natura Edition] (BiblioWoods[Natura][v1.5].jar) 
	UCHIJAAAA	BinnieCore{2.0-pre12} [Binnie Core] (binnie-mods-2.0-pre12.jar) 
	UCHIJAAAA	Botany{2.0-pre12} [Botany] (binnie-mods-2.0-pre12.jar) 
	UCHIJAAAA	ExtraBees{2.0-pre12} [Extra Bees] (binnie-mods-2.0-pre12.jar) 
	UCHIJAAAA	ExtraTrees{2.0-pre12} [Extra Trees] (binnie-mods-2.0-pre12.jar) 
	UCHIJAAAA	Genetics{2.0-pre12} [Genetics] (binnie-mods-2.0-pre12.jar) 
	UCHIJAAAA	CarpentersBlocks{3.3.6} [Carpenter's Blocks] (Carpenter's Blocks v3.3.6 - MC 1.7.10.jar) 
	UCHIJAAAA	ChickenChunks{1.3.4.19} [ChickenChunks] (ChickenChunks-1.7.10-1.3.4.19-universal.jar) 
	UCHIJAAAA	MineFactoryReloaded{1.7.10R2.8.0} [MineFactory Reloaded] (MineFactoryReloaded-[1.7.10]2.8.0-104.jar) 
	UCHIJAAAA	EnderIO{1.7.10-2.2.8.381} [Ender IO] (EnderIO-1.7.10-2.2.8.381.jar) 
	UCHIJAAAA	ThermalFoundation{1.7.10R1.0.0} [Thermal Foundation] (ThermalFoundation-[1.7.10]1.0.0-81.jar) 
	UCHIJAAAA	ThermalExpansion{1.7.10R4.0.1} [Thermal Expansion] (ThermalExpansion-[1.7.10]4.0.1-182.jar) 
	UCHIJAAAA	TConstruct{1.7.10-1.8.5.build957} [Tinkers' Construct] (TConstruct-1.7.10-1.8.5.jar) 
	UCHIJAAAA	ExtraTiC{1.4.5} [ExtraTiC] (ExtraTiC-1.7.10-1.4.5.jar) 
	UCHIJAAAA	FastCraft{1.21} [FastCraft] (fastcraft-1.21.jar) 
	UCHIJAAAA	Baubles{1.0.1.10} [Baubles] (Baubles-1.7.10-1.0.1.10.jar) 
	UCHIJAAAA	Thaumcraft{4.2.3.5} [Thaumcraft] (Thaumcraft-1.7.10-4.2.3.5.jar) 
	UCHIJAAAA	ForgeMultipart{1.1.2.334} [Forge Multipart] (ForgeMultipart-1.7.10-1.1.2.334-universal.jar) 
	UCHIJAAAA	ThaumicTinkerer{unspecified} [Thaumic Tinkerer] (ThaumicTinkerer-2.5-1.7.10-502.jar) 
	UCHIJAAAA	ForbiddenMagic{1.7.10-0.562} [Forbidden Magic] (Forbidden Magic-1.7.10-0.562.jar) 
	UCHIJAAAA	MagicBees{1.7.10-2.3.2} [Magic Bees] (magicbees-1.7.10-2.3.2.jar) 
	UCHIJAAAA	MineFactoryReloaded|CompatAppliedEnergistics{1.7.10R2.8.0} [MFR Compat: Applied Energistics] (MineFactoryReloaded-[1.7.10]2.8.0-104.jar) 
	UCHIJAAAA	MineFactoryReloaded|CompatAtum{1.7.10R2.8.0} [MFR Compat: Atum] (MineFactoryReloaded-[1.7.10]2.8.0-104.jar) 
	UCHIJAAAA	MineFactoryReloaded|CompatBackTools{1.7.10R2.8.0} [MFR Compat: BackTools] (MineFactoryReloaded-[1.7.10]2.8.0-104.jar) 
	UCHIJAAAA	MineFactoryReloaded|CompatBuildCraft{1.7.10R2.8.0} [MFR Compat: BuildCraft] (MineFactoryReloaded-[1.7.10]2.8.0-104.jar) 
	UCHIJAAAA	MineFactoryReloaded|CompatChococraft{1.7.10R2.8.0} [MFR Compat: Chococraft] (MineFactoryReloaded-[1.7.10]2.8.0-104.jar) 
	UCHIJAAAA	MineFactoryReloaded|CompatExtraBiomes{1.7.10R2.8.0} [MFR Compat: ExtraBiomes] (MineFactoryReloaded-[1.7.10]2.8.0-104.jar) 
	UCHIJAAAA	MineFactoryReloaded|CompatForestry{1.7.10R2.8.0} [MFR Compat: Forestry] (MineFactoryReloaded-[1.7.10]2.8.0-104.jar) 
	UCHIJAAAA	MineFactoryReloaded|CompatForgeMicroblock{1.7.10R2.8.0} [MFR Compat: ForgeMicroblock] (MineFactoryReloaded-[1.7.10]2.8.0-104.jar) 
	UCHIJAAAA	MineFactoryReloaded|CompatIC2{1.7.10R2.8.0} [MFR Compat: IC2] (MineFactoryReloaded-[1.7.10]2.8.0-104.jar) 
	UCHIJAAAA	Mystcraft{0.11.13.00} [Mystcraft] (mystcraft-1.7.10-0.11.13.00 (1).jar) 
	UCHIJAAAA	MineFactoryReloaded|CompatMystcraft{1.7.10R2.8.0} [MFR Compat: Mystcraft] (MineFactoryReloaded-[1.7.10]2.8.0-104.jar) 
	UCHIJAAAA	MineFactoryReloaded|CompatProjRed{1.7.10R2.8.0} [MFR Compat ProjectRed] (MineFactoryReloaded-[1.7.10]2.8.0-104.jar) 
	UCHIJAAAA	MineFactoryReloaded|CompatRailcraft{1.7.10R2.8.0} [MFR Compat: Railcraft] (MineFactoryReloaded-[1.7.10]2.8.0-104.jar) 
	UCHIJAAAA	MineFactoryReloaded|CompatSufficientBiomes{1.7.10R2.8.0} [MFR Compat: Sufficient Biomes] (MineFactoryReloaded-[1.7.10]2.8.0-104.jar) 
	UCHIJAAAA	MineFactoryReloaded|CompatThaumcraft{1.7.10R2.8.0} [MFR Compat: Thaumcraft] (MineFactoryReloaded-[1.7.10]2.8.0-104.jar) 
	UCHIJAAAA	MineFactoryReloaded|CompatThermalExpansion{1.7.10R2.8.0} [MFR Compat: Thermal Expansion] (MineFactoryReloaded-[1.7.10]2.8.0-104.jar) 
	UCHIJAAAA	MineFactoryReloaded|CompatTConstruct{1.7.10R2.8.0} [MFR Compat: Tinkers' Construct] (MineFactoryReloaded-[1.7.10]2.8.0-104.jar) 
	UCHIJAAAA	TwilightForest{2.3.7} [The Twilight Forest] (twilightforest-1.7.10-2.3.7.jar) 
	UCHIJAAAA	MineFactoryReloaded|CompatTwilightForest{1.7.10R2.8.0} [MFR Compat: TwilightForest] (MineFactoryReloaded-[1.7.10]2.8.0-104.jar) 
	UCHIJAAAA	MineFactoryReloaded|CompatVanilla{1.7.10R2.8.0} [MFR Compat: Vanilla] (MineFactoryReloaded-[1.7.10]2.8.0-104.jar) 
	UCHIJAAAA	runicdungeons{1.1.5} [Runic Dungeons] (runicdungeons-1.1.5.jar) 
	UCHIJAAAA	simplyjetpacks{1.5.1} [Simply Jetpacks] (SimplyJetpacks-MC1.7.10-1.5.1.jar) 
	UCHIJAAAA	SolarExpansion{1.6a} [Solar Expansion] (SolarExpansion-Basic-1.6a.jar) 
	UCHIJAAAA	thaumcraftneiplugin{1.7.10-1.7} [Thaumcraft NEI Plugin] (thaumcraftneiplugin-1.7.10-1.7.jar) 
	UCHIJAAAA	ThaumicExploration{0.6.0} [Thaumic Exploration] (ThaumicExploration-1.7.10-1.1-37.jar) 
	UCHIJAAAA	ThermalDynamics{1.7.10R1.0.0} [Thermal Dynamics] (ThermalDynamics-[1.7.10]1.0.0-122.jar) 
	UCHIJAAAA	TiCTooltips{1.2.4} [TiC Tooltips] (TiCTooltips-mc1.7.10-1.2.4.jar) 
	UCHIJAAAA	McMultipart{1.1.2.334} [Minecraft Multipart Plugin] (ForgeMultipart-1.7.10-1.1.2.334-universal.jar) 
	UCHIJAAAA	MrTJPCoreMod{1.0.7.14} [MrTJPCore] (MrTJPCore-1.7.10-1.0.7.14-universal.jar) 
	UCHIJAAAA	ForgeMicroblock{1.1.2.334} [Forge Microblocks] (ForgeMultipart-1.7.10-1.1.2.334-universal.jar)

**Comment by BurningChromosome** [comment]
I think this error originates from [707266c](https://github.com/MagicBees/MagicBees/commit/707266cad7b5d0cb5bba66c1dd235e2d42904f98), where the initialization of blocks and items was moved further back in the process.

This later initialization is why the bees do not produce anything. I have tested this by recompiling the code with the initialization moved back to the “init” phase. By using this new jar on my server and client the AM2 bees produce the AM2 items again.
![2015-08-02_02 40 40](https://cloud.githubusercontent.com/assets/13392074/9024228/a74f4aea-38c3-11e5-8b1c-41e99be00021.png)

Furthermore the original error [#58](https://github.com/MagicBees/MagicBees/issues/58) was not alleviated by initializing AM2 items later. This is because the original error was caused by the fact that the flowers “Aum” and “TarmaRoot” are registered by AM2 with capital “A” and “T”.
![2015-08-02_02 50 12](https://cloud.githubusercontent.com/assets/13392074/9024237/e86e1286-38c3-11e5-8180-0168cffe51f6.png)
![2015-08-02_02 50 16](https://cloud.githubusercontent.com/assets/13392074/9024238/edf8d74a-38c3-11e5-9e0d-764d0b3cbc50.png)
I have also tested this by compiling a new jar file.

This last correction also addressed the issue brought up by Tukanen on 10 May where he points out that only cerublossom counts as AM2 flowers.
![2015-08-02_02 41 59](https://cloud.githubusercontent.com/assets/13392074/9024239/08e42794-38c4-11e5-886b-ddd506910b89.png)

**Comment by PrinceRenais** [comment]
Looking at the NEI, I too am having this issue with the 2.4.0 version from Curse Forge (Not the one here on Github, guessing by the version file stating 2.3.5 instead). While it sounds like BurningChromosone has it all figured out, I'm a novice programmer at best. I changed the ArsMagickaHelper as suggested, but I haven't the foggiest idea how to compile this code into a big jar file to put on my private server/clients, and while I'm interested in learning how to do this eventually, my last hour of research shows that I'm missing some understanding. It doesn't help that the resources I've found seem to be intended for Eclipse users, rather than NetBeans (the IDE I use), though perhaps I'm just not familiar with something-or-other. 

If BurningChromosone was correct and the change that removed AM2 Production didn't even fix what it was meant to, I believe it'd be a really simple correction for one of you tech-savvy people to change the load order back in that particular file for and then export a new MagicBees.jar version. I'd certainly appreciate it. For quick reference, I'm talking about ~line 65 of src/main/java/magicbees/main/utils/compat/ArsMagicaHelper.java
